libcudacxx/include/cuda/std/detail/libcxx/include/bit

```diff
@@ -166,165 +166,83 @@ inline _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_BIT_CONSTEXPR int __fallback_po
 
 #ifndef _CCCL_COMPILER_MSVC
 
-inline _LIBCUDACXX_INLINE_VISIBILITY constexpr int __libcpp_ctz(unsigned __x) noexcept
+inline _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_BIT_CONSTEXPR int __libcpp_ctz(uint32_t __x) noexcept
 {
-#  if defined(_CCCL_COMPILER_NVRTC) || (defined(_CCCL_CUDACC_BELOW_11_3))
-#    if defined(_LIBCUDACXX_IS_CONSTANT_EVALUATED) && (_CCCL_STD_VER >= 2014)
+#  if defined(_LIBCUDACXX_IS_CONSTANT_EVALUATED) && (_CCCL_STD_VER >= 2014)
   if (!__libcpp_is_constant_evaluated())
   {
     NV_IF_ELSE_TARGET(
-      NV_IS_DEVICE, (return (!__x) ? sizeof(unsigned) * 8 : __ffs(__x) - 1;), (return __builtin_ctz(__x);))
+      NV_IS_DEVICE, (return (!__x) ? sizeof(uint32_t) * 8 : __ffs(__x) - 1;), (return __builtin_ctz(__x);))
   }
-#    endif // defined(_LIBCUDACXX_IS_CONSTANT_EVALUATED) && (_CCCL_STD_VER >= 2014)
-
+#  endif // defined(_LIBCUDACXX_IS_CONSTANT_EVALUATED) && (_CCCL_STD_VER >= 2014)
   return __binary_ctz32(static_cast<uint64_t>(__x), 0);
-#  else // ^^^ _CCCL_COMPILER_NVRTC || nvcc < 11.3 ^^^ / vvv !_CCCL_COMPILER_NVRTC || nvcc >= 11.3 vvv
-  return __builtin_ctz(__x);
-#  endif // !_CCCL_COMPILER_NVRTC || nvcc >= 11.3
```

Comment:
That looks like a pessimization. We should always call the builtin when possible

Reply:
There's a few problems here:
We value constexpr evaluation over using the correct intrinsic. So C++11 forces us to use the binary search rather than the 'correct' implementation.

Reply:
I've removed the guard based on the existence of `is_constant_evaluated()`.